distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso

```diff
@@ -2045,19 +2045,18 @@ type Table
                 self.select_columns column . if_not_error <| lookup_table.select_columns [from_column, to_column] . if_not_error <|
                     unique = self.column_naming_helper.create_unique_name_strategy
                     unique.mark_used (self.column_names)
-                    unique.mark_used (lookup_table.column_names)
 
                     ## We perform a `merge` into `column`, using a duplicate of `column`
                        as the key column to join with `from_column`.
 
                     duplicate_key_column_name = unique.make_unique "duplicate_key"
                     duplicate_key_column = self.at column . rename duplicate_key_column_name
-                    self_with_duplicate = Table.new ([duplicate_key_column] + self.columns)
+                    self_with_duplicate = self.set duplicate_key_column set_mode=Set_Mode.Add
 
                     ## Create a lookup table with just `to_column` and `from_column`,
                        renamed to match the base table's `column` and its duplicate,
                        respectively.
-                    lookup_table_renamed = Table.new [lookup_table.at from_column . rename duplicate_key_column_name, lookup_table.at to_column . rename column]
+                    lookup_table_renamed = lookup_table.select_columns [from_column, to_column] . rename_columns (Map.from_vector [[from_column, duplicate_key_column_name], [to_column, column]])
```

Comment:
btw. you are doing this select twice... once here and once above

Reply:
Done.